astropy/visualization/wcsaxes/tests/test_misc.py

```diff
@@ -225,19 +225,21 @@ def test_slicing_warnings(ignore_matplotlibrc, tmp_path):
     with warnings.catch_warnings():
         # https://github.com/astropy/astropy/issues/9690
         warnings.filterwarnings("ignore", message=r".*PY_SSIZE_T_CLEAN.*")
-        plt.clf()
-        plt.subplot(1, 1, 1, projection=wcs3d, slices=("x", "y", 2))
-        plt.savefig(tmp_path / "test.png")
+        fig.clear()
+        fig.add_subplot(1, 1, 1, projection=wcs3d, slices=("x", "y", 2))
+        fig.savefig(tmp_path / "test.png")
 
 
 def test_plt_xlabel_ylabel(tmp_path):
     # Regression test for a bug that happened when using plt.xlabel
     # and plt.ylabel with Matplotlib 3.0
+    import matplotlib.pyplot as plt
 
-    plt.subplot(projection=WCS())
+    fig = Figure()
+    ax = fig.add_subplot(projection=WCS())
     plt.xlabel("Galactic Longitude")
     plt.ylabel("Galactic Latitude")
```

Comment:
I think this should be ax.set_ylabel?

Reply:
This is a regression test specifically aimed for pyplot.xlabel so I'm not sure if I should change it. There are others like this one and in general I tried to isolate importing pyplot within these.

Reply:
Ah yes, it might make sense to leave the whole test as using plt in this case?

Reply:
sure, that's much clearer now that I don't have 4 or 5 different failures in CI :)

Reply:
Interestingly, reverting the test to its original form revealed an issue, precisely because `pyplot` manages a global state, effectively encouraging test pollution (which is exactly what I'm refactoring against):

[example logs](https://github.com/astropy/astropy/actions/runs/11649985496/job/32438187987)

<details> <summary> output </summary>

```
=================================== FAILURES ===================================
____________________________ test_plt_xlabel_ylabel ____________________________

tmp_path = PosixPath('/tmp/pytest-of-runner/pytest-0/test_plt_xlabel_ylabel0')

    def test_plt_xlabel_ylabel(tmp_path):
        # Regression test for a bug that happened when using plt.xlabel
        # and plt.ylabel with Matplotlib 3.0
        import matplotlib.pyplot as plt
    
>       plt.subplot(projection=WCS())

../../.tox/py311-test-oldestdeps-alldeps-cov-clocale/lib/python3.11/site-packages/astropy/visualization/wcsaxes/tests/test_misc.py:238: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py311-test-oldestdeps-alldeps-cov-clocale/lib/python3.11/site-packages/matplotlib/pyplot.py:1274: in subplot
    _api.warn_deprecated(
../../.tox/py311-test-oldestdeps-alldeps-cov-clocale/lib/python3.11/site-packages/matplotlib/_api/deprecation.py:96: in warn_deprecated
    warn_external(warning, category=MatplotlibDeprecationWarning)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

message = MatplotlibDeprecationWarning('Auto-removal of overlapping axes is deprecated since 3.6 and will be removed two minor releases later; explicitly call ax.remove() as needed.')
category = <class 'matplotlib._api.deprecation.MatplotlibDeprecationWarning'>

    def warn_external(message, category=None):
        """
        `warnings.warn` wrapper that sets *stacklevel* to "outside Matplotlib".
    
        The original emitter of the warning can be obtained by patching this
        function back to `warnings.warn`, i.e. ``_api.warn_external =
        warnings.warn`` (or ``functools.partial(warnings.warn, stacklevel=2)``,
        etc.).
        """
        frame = sys._getframe()
        for stacklevel in itertools.count(1):  # lgtm[py/unused-loop-variable]
            if frame is None:
                # when called in embedded context may hit frame is None
                break
            if not re.match(r"\A(matplotlib|mpl_toolkits)(\Z|\.(?!tests\.))",
                            # Work around sphinx-gallery not setting __name__.
                            frame.f_globals.get("__name__", "")):
                break
            frame = frame.f_back
>       warnings.warn(message, category, stacklevel)
E       matplotlib._api.deprecation.MatplotlibDeprecationWarning: Auto-removal of overlapping axes is deprecated since 3.6 and will be removed two minor releases later; explicitly call ax.remove() as needed.

../../.tox/py311-test-oldestdeps-alldeps-cov-clocale/lib/python3.11/site-packages/matplotlib/_api/__init__.py:363: MatplotlibDeprecationWarning
```

</details>

adding the following line (as a band aid) avoids the problem
```python
plt.figure()
```

Reply:
I think if we want to keep tests that are specifically about pyplot usage, we'll need to design some pytest fixture to better isolate them from one another. To be frank I have no idea how this kind of issue hasn't popped up more often in the past already.